docs/executor_plugins.md

```diff
@@ -162,7 +171,7 @@ spec:
         - containerPort: 4355
       securityContext:
         runAsNonRoot: true
-        runAsUser: 65534 # nobody
+        runAsUser: 1000
```

Comment:
can you elaborate why this is necessary? I'm not sure if this is necessary for all users, or just your specific environment

Reply:
Not sure if it's a local environmental problem, but leaving that to `65534` the examples never works. I provided more info in the original issue. I'll be happy to be proven wrong.

I use OS X (13.6 (22G120)) and Docker Desktop 4.14.1.

Reply:
Yes you mentioned this in [the issue](https://github.com/argoproj/argo-workflows/issues/13015#issuecomment-2098378193) but you didn't say _why_ there either. That's why I asked here

Reply:
This doesn't explain why it needs `runAsUser: 1000`, but apparently the tests do exactly that: https://github.com/argoproj/argo-workflows/blob/2f15709952075ba208cf45ec459e66a14fa37d02/test/e2e/manifests/plugins/hello-executor-plugin-configmap.yaml#L71 ... not sure why the docs were different 🤔

Reply:
`git blame` shows it was actually added in #7855, not during the original feature #7256

Reply:
I see it was added in [the docs too](https://github.com/argoproj/argo-workflows/blob/2f15709952075ba208cf45ec459e66a14fa37d02/docs/executor_plugins.md?plain=1#L250) in #7688.

I couldn't [find a place in the code](https://github.com/search?q=repo%3Aargoproj%2Fargo-workflows%20runAsUser&type=code) where this was actually enforced though.
The only relevant part seemed to be that the [default Executor `runAsUser` is `1000`](https://github.com/argoproj/argo-workflows/blob/2f15709952075ba208cf45ec459e66a14fa37d02/docs/workflow-controller-configmap.yaml#L189) (partly from #8062).

Reply:
Correct you can not enforce the `runAsUser` value for the `ExecutorPlugin` and you can customise that. I thought that was the whole point of documenting that, comparing to the other init/main agent where the `runAsUser` is implicitly defined. For this particular docker image used in the example that UID does not work, so if someone will copy/paste the example and try to run it would fail. That's the reason I wanted to address it.

Reply:
> I thought that was the whole point of documenting that, comparing to the other init/main agent where the `runAsUser` is implicitly defined.

Honestly I wasn't sure myself 😅 Was trying to hunt that down as well as why it was failing



> For this particular docker image used in the example that UID does not work, so if someone will copy/paste the example and try to run it would fail. That's the reason I wanted to address it.

Ah so it's this particular image that fails with that UID. I was thinking it was a deeper problem 😅 . Thanks for clarifying!